packages/accounts/src/msca/builder.ts

```diff
@@ -92,11 +110,39 @@ export class MSCABuilder {
 
   build<TTransport extends SupportedTransports = Transport>(
     params: BaseSmartAccountParams
-  ): MSCA {
+  ): MSCA<TTransport, ReturnType<typeof pluginManagerDecorator>> {
     const builder = this;
     const { signer, executor, factory } = zCompleteBuilder.parse(builder);
 
-    return new (class extends BaseSmartContractAccount<TTransport> {
+    return new (class DynamicMSCA<
+      TProviderDecorators = ReturnType<typeof pluginManagerDecorator>
+    > extends BaseSmartContractAccount<TTransport> {
```

Comment:
With MSCA, there can be multiple owners, while BaseSmartContractAccount has a notion of a single owner atm. should we update this too general multiowner (a list of smart account signers)? How does multiple signer interface managed in regards to MSCAs?